addons/mail/static/src/chatter/web/mail_composer_form.js

```diff
@@ -50,17 +64,94 @@ export class MailComposerFormRenderer extends formView.Renderer {
+            this.env.dialogData.dismiss = callback;
+        };
+
+        // Update the thread recipient lists:
+        onCloseWizardModal(async () => {
+            const thread = getActiveMailThread();
+            const selectedPartnerIds = this.props.record.data.partner_ids.records.map(
+                (partner) => partner.resId
+            );
+            const partners = await this.orm.searchRead(
+                "res.partner",
+                [["id", "in", selectedPartnerIds]],
+                ["email", "id", "lang", "name"]
+            );
+
+            // Update the recipient lists:
+            thread.suggestedRecipients = thread.suggestedRecipients.map((suggestedRecipient) => {
+                const partner = partners.find(
+                    (partner) =>
+                        partner.id === suggestedRecipient.id ||
+                        partner.email === suggestedRecipient.email
+                );
+                if (partner) {
+                    return {
+                        ...suggestedRecipient,
+                        email: partner.email,
+                        lang: partner.lang,
+                        name: partner.name,
+                        partner_id: partner.id,
+                        persona: { type: "partner", id: partner.id },
+                    };
+                }
+                return suggestedRecipient;
+            });
+            thread.additionalRecipients = thread.additionalRecipients.map((additionalRecipient) => {
+                const partner = partners.find(
+                    (partner) =>
+                        partner.id === additionalRecipient.id ||
+                        partner.email === additionalRecipient.email
+                );
+                if (partner) {
+                    return {
+                        ...additionalRecipient,
+                        email: partner.email,
+                        lang: partner.lang,
+                        name: partner.name,
+                        partner_id: partner.id,
+                        persona: { type: "partner", id: partner.id },
+                    };
+                }
+                return additionalRecipient;
+            });
+
+            // Remove the recipients that got removed from the composer:
+            thread.suggestedRecipients = thread.suggestedRecipients.filter((suggestedRecipient) =>
+                selectedPartnerIds.includes(suggestedRecipient.partner_id)
+            );
+            thread.additionalRecipients = thread.additionalRecipients.filter(
+                (additionalRecipient) => selectedPartnerIds.includes(additionalRecipient.partner_id)
+            );
+
+            // Add the recipients that got added to the composer:
+            for (const partner of partners) {
+                const allRecipients = [
+                    ...thread.suggestedRecipients,
+                    ...thread.additionalRecipients,
```

Comment:
are we sure that no recipient can be present in both aggregates at the same time ?